indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3236,6 +3256,12 @@ private void checkPendingCompletionTasks()
 
   private void checkCurrentTaskState() throws ExecutionException, InterruptedException, TimeoutException
   {
+    List<Task> tasks = taskStorage.getActiveTasksByDatasource(dataSource);
```

Comment:
could we use the list we found in `discoverTasks` itself?

Reply:
`discoverTasks()` is followed by `updateTaskStatus()`, `checkTaskDuration()` and `checkPendingCompletionTasks()`. It's possible that one of them might update the status of these tasks thus changing the set of active tasks. So fetching them again here wouldn't hurt.

Reply:
That said, the set of active tasks can only reduce, not increase. So we could potentially create the map of active tasks in `runInternal` itself and pass that map to both `discoverTasks` and `checkCurrentTaskState`.